contracts/oracles/TwapOracle.sol

```diff
@@ -215,7 +211,7 @@ contract TwapOracle is AccessControlled, TwapInterface {
         (uint256 nowCumulativePrice, uint256 oldCumulativePrice, uint256 oldTimestamp) = pokeWindowValues(config);
 
         // This should be impossible, but better safe than sorry
-        require(block.timestamp > oldTimestamp, "now must come after before");
+        if (block.timestamp <= oldTimestamp) revert("now must come after before");
```

Comment:
I wonder if it's indeed the behavior that we want here. The comment above says that "this should be impossible" but AFAIU this revert could be triggered if TWAP is updated twice in the same block for the same asset.

We do catch all errors here: https://github.com/VenusProtocol/oracle/blob/c5ea1b960089d3b3a617449ed9e778f179c2a75a/contracts/ResilientOracle.sol#L176-L177, so it won't affect the protocol operation, but it looks like _conceptually_ we should handle this "same block" case. I expect it to be quite common actually.